### DISCUSSION TITLE
Bigquery error: AttributeError: type object 'PollingFuture' has no attribute '_DEFAULT_VALUE'

### DISCUSSION THREAD

**Discussion by MrMoronIV** [sqlalchemy/sqlalchemy#12047] [Usage Questions]
On trying to connect:

connection_string f"bigquery://" path is a link to my json file with the credentials.

This seems to be an error inside google's package? I swear it worked a month ago or so.

Traceback (most recent call last):                                                                                                                                                         │
│
│    engine = sqlalchemy.create_engine(connection_string,credentials_path=path)                                                                                                             │
│  File "<string>", line 2, in create_engine                                                                                                                                                │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy/util/deprecations.py", line 281, in warned                                                                               │
│    return fn(*args, **kwargs)  # type: ignore[no-any-return]                                                                                                                              │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy/engine/create.py", line 550, in create_engine                                                                            │
│    entrypoint = u._get_entrypoint()                                                                                                                                                       │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy/engine/url.py", line 758, in _get_entrypoint                                                                             │
│    cls = registry.load(name)                                                                                                                                                              │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py", line 373, in load                                                                                  │
│    return impl.load()                                                                                                                                                                     │
│  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 171, in load                                                                                                             │
│    module = import_module(match.group('module'))                                                                                                                                          │
│  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module                                                                                                             │
│    return _bootstrap._gcd_import(name[level:], package, level)                                                                                                                            │
│  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import                                                                                                                          │
│  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load                                                                                                                       │
│  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked                                                                                                              │
│  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked                                                                                                                        │
│  File "<frozen importlib._bootstrap_external>", line 883, in exec_module                                                                                                                  │
│  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed                                                                                                             │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy_bigquery/__init__.py", line 25, in <module>                                                                              │
│    from .base import BigQueryDialect, dialect                                                                                                                                             │
│  File "/home/odoo/.local/lib/python3.10/site-packages/sqlalchemy_bigquery/base.py", line 30, in <module>                                                                                  │
│    from google.cloud.bigquery import dbapi                                                                                                                                                │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/__init__.py", line 36, in <module>                                                                            │
│    from google.cloud.bigquery.client import Client                                                                                                                                        │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/client.py", line 69, in <module>                                                                              │
│    from google.cloud.bigquery import _job_helpers                                                                                                                                         │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/_job_helpers.py", line 47, in <module>                                                                        │
│    from google.cloud.bigquery import job                                                                                                                                                  │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/job/__init__.py", line 17, in <module>                                                                        │
│    from google.cloud.bigquery.job.base import _AsyncJob                                                                                                                                   │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/job/base.py", line 30, in <module>                                                                            │
│    from google.cloud.bigquery.retry import (                                                                                                                                              │
│  File "/home/odoo/.local/lib/python3.10/site-packages/google/cloud/bigquery/retry.py", line 150, in <module>                                                                              │
│    POLLING_DEFAULT_VALUE = google.api_core.future.polling.PollingFuture._DEFAULT_VALUE                                                                                                    │
│AttributeError: type object 'PollingFuture' has no attribute '_DEFAULT_VALUE'

**Comment by CaselIT** [comment]
Hi.

the bigquery dialect is not maintained here, so you may have more luck by asking in the sqlalchemy_bigquery repository

**Reply by gordthompson** [reply]
https://github.com/googleapis/python-bigquery-sqlalchemy
